### PR TITLE
Add unit tests setup

### LINKED ISSUE #22: Tests, tests and more tests!
We need a test project.  Each test will need to setup/teardown its own sample db. We will need to make this work on the build server somehow maybe some cloud DB (not sql azure)

### PR DESCRIPTION
**Pull request by feanz**
resolves #22 
- Added a unit test project and a test solution folder. 
- Added xunit and shouldfluent
- Updated our connection use to something that we can mock
- Added some basic tests to set the pattern of use (more tests to come)

### CODE REVIEW

File: src/lhm.net/App_Packages/Dapper/SqlMapper.cs [hunk 1]
```diff
@@ -5016,7 +5016,7 @@ public DynamicParameters Output<T>(T target, Expression<Func<T, object>> express
             do
             {
                 // Insert the names in the right order so expression 
-                // "Post.Author.Name" becomes parameter "PostAuthorName"
+                // "Post.Author.Destination" becomes parameter "PostAuthorName"
```
Comment (Amila17) [inline src/lhm.net/App_Packages/Dapper/SqlMapper.cs]: This file shouldn't change.

File: src/lhm.net/App_Packages/LibLog.4.2/LibLog.cs [hunk 2]
```diff
@@ -385,7 +385,7 @@ interface ILogProvider
         /// <summary>
         /// Gets the specified named logger.
         /// </summary>
-        /// <param name="name">Name of the logger.</param>
+        /// <param name="name">Destination of the logger.</param>
```
Comment (Amila17) [inline src/lhm.net/App_Packages/LibLog.4.2/LibLog.cs]: This file shouldn't change.

File: src/lhm.net/Migrator.cs [hunk 3]
```diff
@@ -29,24 +28,32 @@ public Migrator(Table origin, IDbConnection connection)
             _dateTimeStamp = DateTime.UtcNow.ToString(Constants.DateFormat);
         }
 
-        public string Name
+        public string Destination
```
Comment (Amila17) [inline src/lhm.net/Migrator.cs:31]: Should this change?

Comment (feanz) [inline src/lhm.net/Migrator.cs:31]: Yeah I just figured that Migration.Destination was more descriptive than Name.  We could make it LhmTableName what do you think.

Comment (Amila17) [inline src/lhm.net/Migrator.cs:31]: I think LhmTableName sounds better.

**Comment by Amila17** [comment]
:shipit:

**Pull request merged**